### PR TITLE
FEAT Add make_series and make_dataframe

### PR DESCRIPTION
**Pull request by Vincent-Maladiere**
Addresses https://github.com/skrub-data/skrub/pull/784/files#r1358253388

This simple PR implements, for both Pandas and Polars:
- A function to create a dataframe from a dictionary of columns
- A function to create a series from a 1d array

When merged, this can be used directly in https://github.com/skrub-data/skrub/pull/784 for example.

### CODE REVIEW

File: skrub/dataframe/_pandas.py
Comment (jeromedockes) [inline skrub/dataframe/_pandas.py:15]:
```suggestion
def make_dataframe(X, index=None):
```

File: skrub/dataframe/_polars.py
Comment (jeromedockes) [inline skrub/dataframe/_polars.py:36]: would it make sense to raise an error when `index` is not `None`? having an index when we are building a polars dataframe might point to a bug. If not, maybe we could add a `del index` statement to make it clear we don't use it

File: skrub/dataframe/_pandas.py
Comment (jeromedockes) [inline skrub/dataframe/_pandas.py:31]: what kind of mistake do you foresee to make this check necessary?

Comment (Vincent-Maladiere) [reply skrub/dataframe/_pandas.py:31]: It may be slightly overkill, but I think about dictionaries of 2d arrays or dataframes.
For instance:
```python
pd.DataFrame(
	dict(
    	a=[[1, 2, 1]],
	    b=[1, 2, 3],
	)
)
```
Will raise `ValueError: All arrays must be of the same length`, but this error is not informative enough IMO.
I suspect this kind of error might happen more than we'd expect. WDYT?

File: skrub/dataframe/_polars.py
Comment (Vincent-Maladiere) [reply skrub/dataframe/_polars.py:36]: Yes, I think it's reasonable to raise an error when `index` is not `None` for Polars.

File: skrub/dataframe/_pandas.py
Comment (jeromedockes) [reply skrub/dataframe/_pandas.py:31]: but this is a private function, right? that will be used only by skrub code not directly by users. so if skrub passes it incorrect inputs there isn't too much the user can do about it except report the problem

btw WDYT about renaming `dataframe` to `_dataframe`? we want skrub estimators to support both pandas and polars, but I'm not sure we want to publicly expose the machinery that makes it possible

Comment (Vincent-Maladiere) [reply skrub/dataframe/_pandas.py:31]: Ok for the renaming, I'll remove the check

**Comment by Vincent-Maladiere** [comment]
I adjusted the API to make the `dataframe` module private (it seems I still have some work on the examples).
I also did some housekeeping by removing the `DataFrameLike` and `SeriesLike` type hints, LMKWYT :)

**Comment by jeromedockes** [comment]
LGTM, I think once you remove the merge conflict we can merge it

**Comment by jeromedockes** [comment]
I think you also need to update `test_select_cols`

**Comment by jeromedockes** [comment]
and the docstring of `get_df_namespace` also mentions `skrub.dataframe`

**Review by jeromedockes — APPROVED** [review]
LGTM, thanks!

**Pull request merged**
